File: shared/src/api/sqm.ts
export type SQMQdisc = 'cake' | 'fq_codel';

export type SQMScript = 'piece_of_cake.qos' | 'layer_cake.qos' | 'simple.qos';

export interface SQMConfig {
  readonly enabled: boolean;
  readonly interface: string;
  readonly download_kbit: number;
  readonly upload_kbit: number;
  readonly qdisc: SQMQdisc;
  readonly script: SQMScript;
  readonly advanced_hint?: string;
  readonly detected_uci_section?: string;
}

export interface SQMApplyResult {
  readonly ok: boolean;
  readonly output?: string;
  readonly error?: string;
}

File: backend/internal/models/sqm.go
package models

// SQMConfig is a minimal subset of sqm-scripts UCI config for common setups.
// Stored in /etc/config/sqm as a "queue" section.
type SQMConfig struct {
	Enabled       bool   `json:"enabled"`
	Interface     string `json:"interface"`
	DownloadKbit  int    `json:"download_kbit"`
	UploadKbit    int    `json:"upload_kbit"`
	Qdisc         string `json:"qdisc"`
	Script        string `json:"script"`
	AdvancedHint  string `json:"advanced_hint,omitempty"`
	DetectedUCIID string `json:"detected_uci_section,omitempty"`
}

// SQMApplyResult is returned by POST /api/v1/sqm/apply.
type SQMApplyResult struct {
	OK     bool   `json:"ok"`
	Output string `json:"output,omitempty"`
	Error  string `json:"error,omitempty"`
}
